objects/SetupChecker/script.lua

```diff
@@ -1915,17 +1917,26 @@ function gainSpirit(player)
                 if aspect ~= nil and aspect ~= "" then
                     label = label.." - "..aspect
                 end
-                obj.createButton({
-                    click_function = "pickSpirit" .. i,
-                    function_owner = self,
-                    label = label,
-                    position = Vector(0,3,-5 + 2*i),
-                    rotation = Vector(0,0,0),
-                    scale = Vector(2/2.3,1,2/1.42),
-                    width = 4850,
-                    height = 600,
-                    font_size = 275,
-                })
+                if i > numButtons then
+                    obj.createButton({
+                        click_function = "pickSpirit" .. i,
+                        function_owner = self,
+                        label = label,
+                        position = Vector(0,3,-5 + 2*i),
+                        rotation = Vector(0,0,0),
+                        scale = Vector(2/2.3,1,2/1.42),
+                        width = 4850,
+                        height = 600,
+                        font_size = 275,
+                    })
+                else
+                    obj.editButton({
+                        index = i - 1,
+                        label = label,
+                        width = 4850,
```

Comment:
when would width and height here not already be 4850 and 600?

Reply:
After the code in `replaceSpirit` failed to find a spirit (perhaps because filters were stricter than they are now), the size gets set to 0. (I tried first without these, and the text showed up, but the button didn't).

Reply:
ahh yea this makes sense now, thanks